File: Blanquita-Backend/app/Models/Rodela/Rodela.py
from datetime import date
from pydantic import BaseModel


class RodelaItem(BaseModel):
    CodigoRodela: str


class IngresoRodelaRequest(BaseModel):
    IdProveedor: int
    IdTipoRodela: int
    Rodelas: list[RodelaItem]


class IngresoRodelaResponse(BaseModel):
    FechaRecepcion: date
    CantidadRodelas: int


class TipoRodela(BaseModel):
    IdTipoRodela: int
    NombreTipoRodela: str
    Descripcion: str | None = None

File: Blanquita-Frontend/src/models/Rodela/Rodela.js
export const RodelaItem = (codigoRodela) => ({
  CodigoRodela: codigoRodela
});

export const IngresoRodelaRequest = (idProveedor, idTipoRodela, rodelas) => ({
  IdProveedor: idProveedor,
  IdTipoRodela: idTipoRodela,
  Rodelas: rodelas.map(RodelaItem)
});

export const IngresoRodelaResponse = (data) => ({
  FechaRecepcion: data.FechaRecepcion,
  CantidadRodelas: data.CantidadRodelas
});


export const TipoRodelaIngreso = (data) => ({
  IdTipoRodela: data.IdTipoRodela,
  NombreTipoRodela: data.NombreTipoRodela,
  Descripcion: data.Descripcion ?? null
});
